### PR TITLE
Medley of (hopefully) uncontroversial fixups towards building with no exceptions

### PR DESCRIPTION
**Pull request by dalg24**
Partial fix for #7037

### CODE REVIEW

File: algorithms/unit_tests/TestBinSortA.hpp [hunk 1]
```diff
@@ -247,10 +247,10 @@ TEST(TEST_CATEGORY, BinSortEmptyView) {
...
 
   // test all exposed public sort methods
-  ASSERT_NO_THROW(Sorter.sort(ExecutionSpace(), v, 0, 0));
-  ASSERT_NO_THROW(Sorter.sort(v, 0, 0));
-  ASSERT_NO_THROW(Sorter.sort(ExecutionSpace(), v));
-  ASSERT_NO_THROW(Sorter.sort(v));
+  Sorter.sort(ExecutionSpace(), v, 0, 0);
+  Sorter.sort(v, 0, 0);
+  Sorter.sort(ExecutionSpace(), v);
+  Sorter.sort(v);
```
Comment (Rombur) [inline algorithms/unit_tests/TestBinSortA.hpp:253]: Add a comment that we are just checking that these functions don't throw. It's pretty obvious what these tests do when you see `ASSERT_NO_THROW` but once you remove the macro it's not clear anymore. In 6 months, we will wonder why these tests don't check anything.

**Review by crtrott — APPROVED** [review]
It's fine with me, I don't think any of the places changed throwing is particularly useful in the first place.

File: algorithms/unit_tests/TestBinSortA.hpp (hunk 1 above)
Comment (dalg24) [reply algorithms/unit_tests/TestBinSortA.hpp:253]: Done

**Review by Rombur — APPROVED** [review]

**Review by romintomasetti — APPROVED** [review]
I would suggest that you state somewhere explicitly that for `noexcept` support, tests should not be using `ASSERT_NO_THROW` and `EXPECT_NO_THROW` anymore.

Writing it in the description of the PR would already help (for the future anyone wanting to understand why we forbid these 2 test macros).

**Comment by dalg24** [comment]
Ignoring runtime failures (no gpu available for execution) in OpenACC build.  Everything else passed.

**Pull request merged**